src/test/groovy/com/palantir/gradle/graal/GradleGraalEndToEndSpec.groovy

```diff
@@ -42,6 +42,10 @@ class GradleGraalEndToEndSpec extends IntegrationSpec {
             outputName 'hello-world'
             graalVersion '1.0.0-rc5'
         }
+        
+        nativeImage {
+            option '-H:Optimize=1'
```

Comment:
tests should include understanding what happens when provided options conflict with native supported gradle options

Reply:
outputName and mainClass are nativeImage options, not general graal options.

I'd also prefer 'nativeImage' configuration live underneath 'graal', for organization.

Reply:
Not sure I understand your feedback here... what's an example of a native supported gradle option that you'd like to avoid conflicting with?

This PR didn't touch the outputName and mainClass properties, but are you requesting that they now live under the nativeImage block? e.g.:

```gradle
graal {
  graalVersion '1.0.0-rc5'
  nativeImage {
    outputName 'hello-world'
    mainClass 'com.palantir.test.Main'
  }
}
```